hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/rollback/ListingBasedRollbackHelper.java

```diff
@@ -68,107 +67,107 @@ public ListingBasedRollbackHelper(HoodieTableMetaClient metaClient, HoodieWriteC
-   * @param doDelete          {@code true} if deletion has to be done. {@code false} if only stats are to be collected w/o performing any deletes.
+   * @param doDelete          {@code true} if deletion has to be done.
+   *                          {@code false} if only stats are to be collected w/o performing any deletes.
    * @return stats collected with or w/o actual deletions.
    */
-  Map<String, HoodieRollbackStat> maybeDeleteAndCollectStats(HoodieEngineContext context,
-                                                             HoodieInstant instantToRollback,
-                                                             List<ListingBasedRollbackRequest> rollbackRequests,
-                                                             boolean doDelete) {
-    return context.mapToPair(rollbackRequests, rollbackRequest -> {
-      switch (rollbackRequest.getType()) {
-        case DELETE_DATA_FILES_ONLY: {
-          final Map<FileStatus, Boolean> filesToDeletedStatus = deleteBaseFiles(metaClient, config, instantToRollback.getTimestamp(),
-              rollbackRequest.getPartitionPath(), doDelete);
-          return new ImmutablePair<>(rollbackRequest.getPartitionPath(),
-              HoodieRollbackStat.newBuilder().withPartitionPath(rollbackRequest.getPartitionPath())
-                  .withDeletedFileResults(filesToDeletedStatus).build());
-        }
-        case DELETE_DATA_AND_LOG_FILES: {
-          final Map<FileStatus, Boolean> filesToDeletedStatus = deleteBaseAndLogFiles(metaClient, config, instantToRollback.getTimestamp(), rollbackRequest.getPartitionPath(), doDelete);
-          return new ImmutablePair<>(rollbackRequest.getPartitionPath(),
-              HoodieRollbackStat.newBuilder().withPartitionPath(rollbackRequest.getPartitionPath())
-                  .withDeletedFileResults(filesToDeletedStatus).build());
-        }
-        case APPEND_ROLLBACK_BLOCK: {
-          String fileId = rollbackRequest.getFileId().get();
-          String latestBaseInstant = rollbackRequest.getLatestBaseInstant().get();
-
-          // collect all log files that is supposed to be deleted with this rollback
-          Map<FileStatus, Long> writtenLogFileSizeMap = FSUtils.getAllLogFiles(metaClient.getFs(),
-              FSUtils.getPartitionPath(config.getBasePath(), rollbackRequest.getPartitionPath()),
-              fileId, HoodieFileFormat.HOODIE_LOG.getFileExtension(), latestBaseInstant)
-              .collect(Collectors.toMap(HoodieLogFile::getFileStatus, value -> value.getFileStatus().getLen()));
-
-          HoodieLogFormat.Writer writer = null;
+  private Pair<String, HoodieRollbackStat> maybeDeleteAndCollectStats(ListingBasedRollbackRequest rollbackRequest,
+                                                                      HoodieInstant instantToRollback,
+                                                                      boolean doDelete) throws IOException {
+    switch (rollbackRequest.getType()) {
```

Comment:
I assume you have not changed anything within this method.

Reply:
I only changed the return type so that it can be reused across different engines.